compiler/modules/front.syntax/token.bal

```diff
@@ -275,6 +285,30 @@ class Tokenizer {
         return err:syntax(msg, loc=err:location(self.file, self.currentPos()));
     }
 
+    function save() returns TokenizerState {
+        return { lineIndex : self.lineIndex,
+                 fragCodeIndex : self.fragCodeIndex,
+                 codePointIndex : self.codePointIndex,
+                 fragmentIndex : self.fragmentIndex,
+                 tokenStartCodePointIndex : self.tokenStartCodePointIndex,
+                 mode : self.mode,
+                 curTok : self.curTok.cloneReadOnly() };
```

Comment:
Better to make Token be readonly, which requires changing VariableLengthToken.

Reply:
fixed